src/plugins/vis_type_table/public/utils/add_percentage_col.test.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { addPercentageCol } from './add_percentage_col';
+import { FormattedColumn } from '../types';
+import { Table } from '../table_vis_response_handler';
+
+const mockDeserialize = jest.fn(() => ({}));
+jest.mock('../services', () => ({
+  getFormatService: jest.fn(() => ({
+    deserialize: mockDeserialize,
+  })),
+}));
+
+let formattedColumns: FormattedColumn[];
+const rows = [
+  { 'col-0-2': 'Alice', 'col-1-1': 3 },
+  { 'col-0-2': 'Anthony', 'col-1-1': 1 },
+  { 'col-0-2': 'Timmy', 'col-1-1': 1 },
+] as Table['rows'];
+
+beforeEach(() => {
+  formattedColumns = [
+    {
+      id: 'col-0-2',
+      title: 'name.keyword: Descending',
+      formatter: {},
+      filterable: true,
+    },
+    {
+      id: 'col-1-1',
+      title: 'Count',
+      formatter: {},
+      filterable: false,
+      sumTotal: 5,
+    },
+  ] as FormattedColumn[];
+});
+
+describe('addPercentageCol', () => {
+  it('should add new percentage column', () => {
+    const result = addPercentageCol(formattedColumns, 'count', rows, 1);
+    expect(result).toEqual({
+      cols: [
+        {
+          id: 'col-0-2',
+          title: 'name.keyword: Descending',
+          formatter: {},
+          filterable: true,
+        },
+        {
+          id: 'col-1-1',
+          title: 'Count',
+          formatter: {},
+          filterable: false,
+          sumTotal: 5,
+        },
+        {
+          title: 'count percentages',
```

Comment:
Out of scope, but it's a little odd that this function allows the `title` of the percentage column to be set independently from the title of the source column.